pkg/lwip/contrib/sock/ip/lwip_sock_ip.c

```diff
@@ -189,36 +196,137 @@ ssize_t sock_ip_recv_aux(sock_ip_t *sock, void *data, size_t max_len,
     ssize_t res, ret = 0;
     bool nobufs = false;
 
+    bool peek = false;
+    if (aux != NULL) {
+        if (IS_USED(MODULE_SOCK_AUX_PEEK)) {
+            peek = aux->flags & SOCK_AUX_PEEK;
+        }
+    }
+
     assert((sock != NULL) && (data != NULL) && (max_len > 0));
+    const void *old_ctx = ctx;
     while ((res = sock_ip_recv_buf_aux(sock, &pkt, (void **)&ctx, timeout,
                                        remote, aux)) > 0) {
+
+        /* sock_udp_recv_buf() is supposed to return multiple packet fragments
+         * when called multiple times with the same context.
+         * In practise, this is not implemented and it will always return a pointer
+         * to the whole packet on the first call and NULL on the second call, which
+         * releases the packet.
+         * This assertion will trigger should the behavior change in the future.
+         */
+        if (old_ctx) {
+            assert(res == 0 && ctx == NULL);
```

Comment:
args so @miri64 pointed out that it's actually LWIP that makes use of that and that comment is wrong :grimacing: